packages/bpk-component-calendar/src/custom-proptypes.js

```diff
@@ -18,20 +18,43 @@
 
 import PropTypes from 'prop-types';
 
-const CalendarConfigurationSingle = PropTypes.shape({
-  type: 'single',
+const startDateType = props => {
+  if (props.startDate > props.endDate) {
```

Comment:
I think we also need to explicitly check that they are both defined and handle that too? 

Further, does this get passed only the sub-object  passed as `selectionConfiguration` not the whole props object? 

We might consider adding some tests for this too

Reply:
Yeah I am not sure what you were getting at with this custom type so was my stab at implementing it. 

> Further, does this get passed only the sub-object passed as selectionConfiguration not the whole props object?

Good question I will check but just reused this thinking from other components, as I am not 100% sure how you would pass just the `selectionConfiguration` option through from the props

would it be `  startDate: startDateType(props.selectionConfiguration.startDate),`? As I don't think it would have specific knowledge of the particular props in this file at runtime

Reply:
Also I have also realised the problem of 

`props.startDate > props.endDate` should actually be `props.selectionConfiguration.startDate > props.selectionConfiguration.endDate` as these dates are in that object

Reply:
If I am honest with you making a custom type like this came as a suprise as this seems more like logic I was going to be implementing for the range itself vs doing it at the earlier prop checking state

As how would this handle if the consumer selected endDate first instead of startDate and the value is null?

Reply:
Something like this is what I am thinking

```diff
diff --git a/packages/bpk-component-calendar/src/custom-proptypes.js b/packages/bpk-component-calendar/src/custom-proptypes.js
index 56f522a39..7988b25e7 100644
--- a/packages/bpk-component-calendar/src/custom-proptypes.js
+++ b/packages/bpk-component-calendar/src/custom-proptypes.js
@@ -18,13 +18,35 @@
 
 import PropTypes from 'prop-types';
 
-const startDateType = props => {
-  if (props.startDate > props.endDate) {
+import { isBefore, isSameDay } from './date-utils';
+
+const startDateType = key => props => {
+  const { startDate, endDate } = props[key];
+
+  // No range selected
+  if (!startDate && !endDate) {
     return null;
   }
-  return new Error(
-    `Start date \`${props.startDate}\` cannot be after end date \`${props.endDate}\`.`,
-  );
+
+  // End date with not start date is not allowed
+  if (!startDate && endDate) {
+    return new Error(
+      `Cannot specify \`endDate\` without \`startDate\` in ${key}.`,
+    );
+  }
+
+  // Start date without an end date is always valid
+  if (startDate && !endDate) {
+    return null;
+  }
+
+  if (isBefore(endDate, startDate) && !isSameDay(endDate, startDate)) {
+    return new Error(
+      `Start date \`${startDate}\` cannot be after end date \`${endDate}\` in \`props.${key}\`.`,
+    );
+  }
+
+  return null;
 };
 
 const endDateType = props => {
@@ -48,7 +70,7 @@ const SelectionConfigurationSingle = PropTypes.shape({
 
 const SelectionConfigurationRange = PropTypes.shape({
   type: CALENDAR_SELECTION_TYPE.range,
-  startDate: startDateType,
+  startDate: startDateType('selectionConfiguration'),
   endDate: endDateType,
 });
 
diff --git a/packages/bpk-component-calendar/src/date-utils.js b/packages/bpk-component-calendar/src/date-utils.js
index 9c7e04a2e..e567b7fea 100644
--- a/packages/bpk-component-calendar/src/date-utils.js
+++ b/packages/bpk-component-calendar/src/date-utils.js
@@ -195,6 +195,7 @@ export {
   getMonthRange,
   getDay,
   dateToBoundaries,
+  isBefore,
   isWithinRange,
   isSaturday,
   isSunday,

```

Reply:
> As how would this handle if the consumer selected endDate first instead of startDate and the value is null?
This isn't allowed I think. If you have no selection or have already selected a range the next selection should cause the `startDate` to be set.